weather_dl/download_pipeline/clients.py

```diff
@@ -209,6 +215,101 @@ def num_requests_per_key(self, dataset: str) -> int:
         return 2
 
 
+class EumetsatClient(Client):
+    def __init__(self, config: Config, level: int = logging.INFO) -> None:
+        super().__init__(config, level)
+        self.key = config.kwargs.get('api_key', os.environ.get("EUMETSATAPI_KEY"))
+        self.secret = config.kwargs.get('api_secret', os.environ.get("EUMETSATAPI_SECRET"))
+
+    def products(self, time_slices: t.List[t.Tuple[np.datetime64, np.datetime64]], dataset: str) -> t.List:
+        """Lists the products that are available for that dataset in the date ranges.
+
+        Args:
+            time_slices: list of tuples of start and end time.
+            dataset: dataset name as listed in eumetsat catalogue.
+        Returns:
+            list of products within the specified times.
+        """
+        datastore = eumdac.DataStore(self._token())
+        collection = datastore.get_collection(dataset)
+        product_list = []
+        for start, end in time_slices:
+            product_list.extend([product._id for product in collection.search(dtstart=start, dtend=end)])
+        return product_list
+
+    def _token(self) -> eumdac.AccessToken:
+        credentials = (self.key, self.secret)
+        return eumdac.AccessToken(credentials)
+
+    def download_native(self, product: eumdac.product.Product, output: str) -> None:
+        """Downloads the entry file from the product."""
+        entry = f'{product}.nat'
+        with product.open(entry=entry) as fsrc:
+            with open(output, 'wb') as fdst:
+                shutil.copyfileobj(fsrc, fdst, DEFAULT_READ_BUFFER_SIZE)
+
+    def download_custom(self, product: eumdac.product.Product, token: eumdac.AccessToken,
+                        chain_config: eumdac.tailor_models.Chain, output: str) -> None:
+        """Downloads the prduct after customisation."""
+        datatailor = eumdac.DataTailor(token)
+        with datatailor.new_customisation(product, chain_config) as customisation:
+            customisation.update_margin = 0
+            timeout = 900  # customisation can be slow
+            tic = time.time()
+            while time.time() - tic < timeout:
+                if customisation.status == 'DONE':
+                    break
+            else:
+                raise TimeoutError(f'Customisation took longer than {timeout}s')
+            self.logger.info('Customisation for product %s: output %s', product, customisation.outputs)
+            with customisation.stream_output(customisation.outputs[0]) as stream:
+                with open(output, 'wb') as fdst:
```

Comment:
Have you tested that this works? I initially managed to download some files directly in netcdf using this, but recently it has failed with various errors. I think unless this works reliably, we should just support downloading native.